src/hotspot/os/linux/os_linux.cpp

```diff
@@ -3658,17 +3658,16 @@ size_t os::Linux::find_default_large_page_size() {
   return large_page_size;
 }
 
-size_t os::Linux::find_large_page_size(size_t large_page_size) {
-  if (_default_large_page_size == 0) {
-    _default_large_page_size = Linux::find_default_large_page_size();
-  }
-  // We need to scan /sys/kernel/mm/hugepages
+void os::Linux::register_large_page_sizes() {
+  // Scan /sys/kernel/mm/hugepages
   // to discover the available page sizes
   const char* sys_hugepages = "/sys/kernel/mm/hugepages";
 
   DIR *dir = opendir(sys_hugepages);
   if (dir == NULL) {
-    return _default_large_page_size;
+    // If we can't open /sys/kernel/mm/hugepages
+    // Add _default_large_page_size to _page_sizes
+    _page_sizes.add(_default_large_page_size);
```

Comment:
missing return here.

Reply:
But see my general remarks. I would modify this function to not change outside state at all, just to return the found page sizes in a os::PageSizes object.